src/result.js

```diff
@@ -112,22 +112,34 @@ class Result {
   }
 }
 
-/**
- * `String.replace()` converts '$$' to '$', so we must escape each '$' as '$$';
- * but because we’re using `String.replace()` to do it, we must use '$$$'!
- */
-function escapeForStringReplace(string) {
-  return string.replace(/\$/g, '$$$');
-}
-
 function insertIntoIndexHTML(html, head, body) {
-  html = html.replace("<!-- EMBER_CLI_FASTBOOT_BODY -->", escapeForStringReplace(body));
+  if (!html) { return Promise.resolve(html); }
+
+  if (body) {
+    let isBodyReplaced = false;
+    html = html.replace("<!-- EMBER_CLI_FASTBOOT_BODY -->", function() {
+      isBodyReplaced = true;
```

Comment:
I was more curious to understand the motivation behind the change. Did you hit this usecase where the comments weren't present?